pkg/dockerregistry/server/auth.go

```diff
@@ -295,10 +320,19 @@ func (ac *AccessController) Authorized(ctx context.Context, accessRecords ...reg
 	}
 
 	// In case of docker login, hits endpoint /v2
-	if len(accessRecords) == 0 {
-		if err := verifyOpenShiftUser(ctx, osClient); err != nil {
+	if len(bearerToken) > 0 {
+		if user, userid, err := verifyOpenShiftUser(ctx, osClient); err != nil {
 			return nil, ac.wrapErr(ctx, err)
+		} else {
+			ctx = WithUserInfoLogger(ctx, user, userid)
 		}
+	} else {
+		ctx = WithUserInfoLogger(ctx, defaultUserName, "")
```

Comment:
Previously, when `len(accessRecords) == 0 && len(bearerToken) == 0`, this function failed with a challenge. Now it passes and I doubt we want that.

Nevertheless, this branch won't ever be executed thanks to `getOpenShiftAPIToken()` token above.